controllers/apps/transformer_cluster_component.go

```diff
@@ -811,6 +811,25 @@ func (h *clusterShardingHandler) updateComps(transCtx *clusterTransformContext,
 }
 
 func (h *clusterShardingHandler) protoComps(transCtx *clusterTransformContext, name string, running *appsv1.Component) ([]*appsv1.Component, error) {
+	buildAnnotations := func(shardingName, compName string) map[string]string {
+		var annotations map[string]string
+		if transCtx.annotations != nil {
+			annotations = transCtx.annotations[compName]
+		}
```

Comment:
annotations := transCtx.annotations[compName].  golang will return empty value even if the map is nil